packages/amplify-graphql-relational-transformer/src/resolvers.ts

```diff
@@ -8,6 +8,57 @@ import { getSortKeyFields } from './schema';
 import { HasManyDirectiveConfiguration } from './types';
 import { getConnectionAttributeName, getObjectPrimaryKey } from './utils';
 
+export const updateTableForReferencesConnection = (
+  config: HasManyDirectiveConfiguration, // TODO: Add support for HasOneDirectiveConfiguration
+  ctx: TransformerContextProvider
+): void => {
+  const { field, fieldNodes, indexName: incomingIndexName, object, references, relatedType } = config;
+
+  if (incomingIndexName) {
+    // TODO: log warning or throw that indexName isn't supported for DDB references
+    // Ideally validate this further up the chain.
+  }
+
+  if (references.length < 1) {
+    throw new Error( // TODO: better error message
+      'references should not be empty here'
+    )
+  }
+
+  const mappedObjectName = ctx.resourceHelper.getModelNameMapping(object.name.value);
```

Comment:
I believe the `getModelNameMapping` is handling models renamed with `mapsTo`, but since we're trying to construct a GSI name for the related table, we should be looking for the name of the related model, not the primary model, right?

Reply:
> but since we're trying to construct a GSI name for the related table

Yes, this is the case.

> we should be looking for the name of the related model, not the primary model, right?

No, I believe it's correct as-is -- we should be using the `<Primary>.<Primary.FieldName>`. I'll break down what's happening here with an example.

Given the following schema.
```graphql
type Team @model {
  id: ID!
  name: String!
  members: [Member] @hasMany(references: ["teamID"])
}

type Member @model {
  id: ID!
  teamID: ID!
  team: Team @belongsTo(references: ["teamID"])
}
```

`mappedObjectName` is `Team` (Primary model) as you mentioned. This leads to a GSI with the name `gsi-Team.members`.

The `gsi-Team.members` GSI is used by the AppSync resolver when querying `Team` and including `members` in the selection set. For example:
```graphql
# Request
query Foo {
  getTeam(id: "001") {
    id
    name
    members { # invokes field resolver, querying `Member` table for `teamID` using `gsi-Team.members`
      items {
        id
        teamID
      }
    }
  }
}

# Response
{
  "data": {
    "getTeam": {
      "id": "001",
      "name": "Inter Miami",
      "members": {
        "items": [
          {
            "id": "12345678-1234-5678-1234-567812345678",
            "name": "Lionel Messi",
            "teamID": "001"
          }
        ]
      }
    }
  }
}
```

This GSI naming convention here makes sense considering it's describing the access pattern for querying the relateds (`members` in this example) through the primary (`Team` in this example). Additionally it's the naming convention used by DDB `fields` based relationships today, so unless it were egregiously incorrect, I'd be reluctant to change it.